packages/core/src/lib/filter/index.ts

```diff
@@ -110,7 +110,7 @@ class Subscription {
           async (source) => await all(source)
         );
 
-        if (!res || !res.length) {
+        if (!res || !res.length || !res[0]) {
```

Comment:
If we can guarantee that `res[0]` will never be a falsy value when defined, then this condition can be shortened to
```
if (!res?.[0]) {
```